internal/store/postgres/resource_repository.go

```diff
@@ -213,3 +214,52 @@ func (r ResourceRepository) GetByURN(ctx context.Context, urn string) (resource.
 
 	return resourceModel.transformToResource(), nil
 }
+
+func buildGetResourcesByNamespaceQuery(dialect goqu.DialectWrapper, withResourceType bool) (string, error) {
+	namespaceQueryExpression := goqu.Ex{
+		"backend": goqu.L("$2"),
+	}
+
+	if withResourceType {
+		namespaceQueryExpression["resource_type"] = goqu.L("$3")
+	}
+
+	getNamespaceQuery := dialect.Select("id").From(TABLE_NAMESPACES).Where(namespaceQueryExpression)
+	getResourcesByURNQuery, _, err := dialect.Select(&ResourceCols{}).From(TABLE_RESOURCES).Where(goqu.Ex{
+		"name":         goqu.L("$1"),
+		"namespace_id": goqu.Op{"in": getNamespaceQuery},
+	}).ToSQL()
+
+	return getResourcesByURNQuery, err
+}
```

Comment:
We can pass and assign the actual variables instead so we don't need to use if-else in GetContext later

```suggestion
func buildGetResourcesByNamespaceQuery(dialect goqu.DialectWrapper,  name string, ns namespace.Namespace) (string, []interface{}, error) {
	getNamespaceQuery := dialect.Select("id").From(TABLE_NAMESPACES)
	getNamespaceQuery = getNamespaceQuery.Where(goqu.Ex{
		"backend": ns.Backend),
	})
	if ns.ResourceType != "" {
	    getNamespaceQuery = getNamespaceQuery.Where(goqu.Ex{
		  "resource_type": ns.ResourceType),
	    })
	}
	
	getResourcesByURNQuery, params, err := dialect.Select(&ResourceCols{}).From(TABLE_RESOURCES).Where(goqu.Ex{
		"name":         name,
		"namespace_id": goqu.Op{"in": getNamespaceQuery},
	}).ToSQL()

	return getResourcesByURNQuery, params, err
}
```

Reply:
resolved